openedx/features/content_tagging/tasks.py

```diff
@@ -0,0 +1,41 @@
+"""
+Defines asynchronous celery task for auto-tagging content
+"""
+
+import logging
+from celery import shared_task
+from celery_utils.logged_task import LoggedTask
+from edx_django_utils.monitoring import set_code_owner_attribute
+from openedx_tagging.core.tagging.models import Taxonomy
+from opaque_keys.edx.keys import CourseKey, UsageKey
+from opaque_keys.edx.locator import LibraryLocator
+from organizations.models import Organization
+
+from . import api
+
+LANGUAGE_TAXONOMY_ID = -1
+ORGANIZATION_TAXONOMY_ID = -3 # -2
+AUTHOR_TAXONOMY_ID = -2 # -3
+
+log = logging.getLogger(__name__)
+
+@shared_task(base=LoggedTask)
+@set_code_owner_attribute
+def update_xblock_tags(usage_key_str: str):
+    """
+    Updates the tags for a XBlock.
+
+    Params:
+        usage_key_str (str): identifier of the XBlock
+    """
+    usage_key = UsageKey.from_string(usage_key_str)
+
+    log.info("Updating tags for XBlock with id: %s", usage_key)
+
+    org = Organization.objects.get(name=usage_key.org)
+    org_taxonomy = Taxonomy.objects.get(pk=ORGANIZATION_TAXONOMY_ID)
+
+    breakpoint()
+    tags = api.get_tags(org_taxonomy) # tags: []
+    api.tag_content_object(org_taxonomy, [org.pk], usage_key) # Error: ValueError: Invalid object tag for taxonomy (-3): 1
```

Comment:
@ChrisChV I'm having `ValueError: Invalid object tag for taxonomy (-3): 1`.
`-3` is the Id of the Org Taxonomy and `1` is the pk of the Org.

Reply:
It seems that the log does not help much to know exactly why the object tag is invalid. @pomegranited Maybe we can improve this in another task later?

Reply:
Sure,  @ChrisChV  we can improve the message.

But @rpenido did you figure out why `tag_content_object` isn't working? I suspect it's because of a bug in the automatic tag creation part of the system taxonomy code?

Reply:
Yes! The problem was that the Taxonomy was not associated with any org. I created some [fixtures](https://github.com/openedx/edx-platform/pull/32907/files#diff-6391549a060d490c43d6021518ba26829f08c45133093bf28a2e2f627fc7f10fR35-R40) to set these taxonomies as (org=None) by default.

Reply:
Perfect! That's exactly the way to fix it @rpenido , thank you for working that out.

Could you ensure the fixtures cover all the system taxonomies we've created here?